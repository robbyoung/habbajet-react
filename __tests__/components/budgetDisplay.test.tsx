import React from 'react';
import renderer from 'react-test-renderer';
import BudgetDisplay from '../../app/components/budgetDisplay';
import {render, fireEvent, wait} from '@testing-library/react-native';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: '',
}));

describe('Budget Display Component', () => {
    it('can render a budget display for the home screen', () => {
        const component = renderer.create(
            <BudgetDisplay budget="$56.43" onPress={() => undefined} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('will run the onPress callback if the button is pressed', async () => {
        const onPress = jest.fn();
        const {getByTestId} = render(
            <BudgetDisplay budget="$80.00" onPress={onPress} />,
        );

        const arrowButton = getByTestId('button-arrow');
        fireEvent.press(arrowButton);
        await wait(() => expect(onPress).toBeCalled());
    });
});
