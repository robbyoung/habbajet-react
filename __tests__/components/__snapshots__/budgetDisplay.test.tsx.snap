// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Budget Display Component can render a budget display with buttons 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Abel",
            "fontSize": 20,
            "margin": 0,
            "padding": 0,
          },
          Object {
            "color": "#959595",
          },
        ]
      }
    >
      My Budget
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Abel",
            "marginTop": -5,
            "padding": 0,
          },
          Object {
            "fontSize": 50,
          },
        ]
      }
    >
      $212.12
    </Text>
  </View>
  <View
    style={
      Object {
        "marginLeft": 15,
        "marginTop": 30,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="button-budget-0"
    >
      <
        color="#959595"
        icon={
          Object {
            "icon": Array [
              640,
              512,
              Array [],
              "f236",
              "M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z",
            ],
            "iconName": "bed",
            "prefix": "fas",
          }
        }
        size={30}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginLeft": 15,
        "marginTop": 30,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="button-budget-1"
    >
      <
        color="#959595"
        icon={
          Object {
            "icon": Array [
              320,
              512,
              Array [],
              "f644",
              "M296 256h-44.62C272.46 222.01 288 181.65 288 144 288 55.63 230.69 0 160 0S32 55.63 32 144c0 37.65 15.54 78.01 36.62 112H24c-13.25 0-24 10.74-24 24v32c0 13.25 10.75 24 24 24h96v152c0 13.25 10.75 24 24 24h32c13.25 0 24-10.75 24-24V336h96c13.25 0 24-10.75 24-24v-32c0-13.26-10.75-24-24-24zM160 80c29.61 0 48 24.52 48 64 0 34.66-27.14 78.14-48 100.87-20.86-22.72-48-66.21-48-100.87 0-39.48 18.39-64 48-64z",
            ],
            "iconName": "ankh",
            "prefix": "fas",
          }
        }
        size={30}
      />
    </View>
  </View>
</View>
`;

exports[`Budget Display Component can render a budget display without buttons 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Abel",
            "fontSize": 20,
            "margin": 0,
            "padding": 0,
          },
          Object {
            "color": "#959595",
          },
        ]
      }
    >
      My Budget
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Abel",
            "marginTop": -5,
            "padding": 0,
          },
          Object {
            "fontSize": 50,
          },
        ]
      }
    >
      $56.43
    </Text>
  </View>
</View>
`;
